File: 2022/july/10-07-2022/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp, length = {}, len(cost)

        def getResult(index):
            if index >= length:
                return 0
            if index in dp:
                return dp[index]

            dp[index] = cost[index] + min(getResult(index + 1), getResult(index + 2))
            return dp[index]

        getResult(0)
        return min(dp[0], dp[1])

File: 2022/july/10-07-2022/min-cost-climbing-stairs.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut one = 0;
        let mut two = 0;
        for i in 2..cost.len()+1{
            let tmp = one;
            one = (one + cost[i-1]).min(two + cost[i-2]);
            two = tmp;
        }
        one
    }
}
